includes/class-wcpdf-install.php

```diff
@@ -177,6 +177,14 @@ protected function install() {
 
 		// set transient for wizard notification
 		set_transient( 'wpo_wcpdf_new_install', 'yes', DAY_IN_SECONDS * 2 );
+
+		// 2.8.2: migrate number stores
+		if ( empty( get_option( 'wpo_wcpdf_number_stores_migrated' ) ) ) {
+			$debug_settings = get_option( 'wpo_wcpdf_settings_debug', array() );
+			$debug_settings['migrate_number_stores'] = 1;
+			update_option( 'wpo_wcpdf_settings_debug', $debug_settings );
+			update_option( 'wpo_wcpdf_number_stores_migrated', 1 );
+		}
```

Comment:
This is now the wrong way around. This will enable the migration for _new installs_ (which have nothing to migrate) and not do any migration for the people that updated (that need their old tables migrated)! and `wpo_wcpdf_number_stores_migrated` will always be empty here since it's a fresh install and will never run again after that.

Instead, you should set the setting to 0 in the debug defaults here:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/70d8cc08f9a6bfbc2fdabf6046fe19c7703ba3f6/includes/class-wcpdf-install.php#L165-L172

Then you can set the default of the setting itself to 1 and also enable `store_unchecked` like you did here:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/70d8cc08f9a6bfbc2fdabf6046fe19c7703ba3f6/includes/documents/class-wcpdf-invoice.php#L309-L310

Finally, when reading the setting, you default to 1 in case no value was set, something like this:
```php
// migrate when setting not set (means default = set) or when set and not 0
$migrate_number_stores = !isset( WPO_WCPDF()->settings->debug_settings['migrate_number_stores'] ) ) || WPO_WCPDF()->settings->debug_settings['migrate_number_stores'] != 0 );

// migrate old stores without year suffix
if( apply_filters( 'wpo_wcpdf_migrate_number_stores', $migrate_number_stores ) ) {
```

but please test thoroughly... And I may be overcomplicating things so if you have a simpler solution, feel free.

The bottom line should be: if anything is not set properly for whatever reason, migration should always be enabled by default.